tutorials/Python/11_OOP/OOP.ipynb

```diff
@@ -0,0 +1,769 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In Python, _everything_ is an object and therefore the concepts of object-oriented programming are in extensive use in every Python code. So, why are objects so widespread? Because they are very useful to write re-usable code and hide away a lot of complexity."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Molecule"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's suppose, for example, that we want to represent a molecule in a Python program. At the very least, we will need to store some data representing the molecule. In order to represent a particular configuration of acetylene, we could store the element number and the atomic coordinates of the atoms in the molecule:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "elements = np.array([1, 6, 6, 1])\n",
+    "coordinates = np.array([\n",
+    "    [-1.06, 0.00, 0.00], # H1\n",
+    "    [ 0.00, 0.00, 0.00], # C1\n",
+    "    [ 1.20, 0.00, 0.00], # C2\n",
+    "    [ 2.26, 0.00, 0.00], # H2\n",
+    "])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Coordinates are stored in a $3 \\times N$ array ($N$ being the number of atoms) and each row is assumed to be ordered in the same way as elements are."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<center>\n",
+    "<img src=\"acetylene.png\" alt=\"acetylene\" width=\"50%\"/>\n",
+    "</center>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This is all good, but we'll need to keep track of the fact that the `elements` and `coordinates` vectors represent the same molecule and therefore should be, in most cases, processed together. And what if we want to represent another molecule? To represent a dihydrogen molecule in the same program we will need to do something like the following:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "elements_H2 = np.array([1, 1])\n",
+    "coordinates_H2 = np.array([\n",
+    "    [0.00, 0.00, 0.00], # H1\n",
+    "    [1.00, 0.00, 0.00], # H1\n",
+    "])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This works, but it's ugly. It is also error-prone and probably will not scale well to many molecules."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> Can you think of a way of tying together `elements` and `coordinates` within a single variable with what you learned already?"
```

Comment:
The formatting here is a bit different from how we've done questions in the other notebooks. It probably doesn't matter, but if you want to go for consistency, maybe just a header or bold entry with a question mark would be sufficient?

Reply:
I thought about this, but making a numbered section for an exercise in a middle of another paragraph is a bit odd in this case. I think in other parts the exercises were at the end of a paragraph anyway but here is more a matter of "stop for a second and have a think". I can definitely change it to make it more consistent, but I'm not sure what's the best way of doing that.

Reply:
I was thinking of just doing ### instead of >, but honestly if you think it looks better this way it's perfectly fine. We can be picky about layout in 2021 :P